clients/src/main/java/org/apache/kafka/common/internals/PartitionStates.java

```diff
@@ -43,11 +43,11 @@
  */
 public class PartitionStates<S> {
 
-    private final LinkedHashMap<TopicPartition, S> map = new LinkedHashMap<>();
+    private final Map<TopicPartition, S> map = new LinkedHashMap<>();
```

Comment:
I think we left the type as `LinkedHashMap` to make the ordering expectation explicit.